### PR TITLE
Cleanup delete schema

### PR DESCRIPTION
**Pull request by drew-gross**
I was going to do more in this PR but there is a dependency on count so I'm going to break that dependency before going further with this.

### CODE REVIEW

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js [hunk 1]
```diff
@@ -81,15 +84,24 @@ export class MongoStorageAdapter {
...
-    }).then(collections => {
+  // Delete all data known to this adatper. Used for testing.
+  deleteAllSchemas() {
+    return this._allCollections().then(collections => {
+      let promises = collections.map(collection => collection.drop());
+      return Promise.all(promises);
+    });
+  }
+
+  _allCollections() {
```
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: can we make it a function with `this` passed as a prop like:

`function allCollections(storageAdapter) { ... }`

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js [hunk 2]
```diff
@@ -81,15 +84,24 @@ export class MongoStorageAdapter {
     });
   }
 
-  // Used for testing only right now.
-  allCollections() {
-    return this.connect().then(() => {
-      return this.database.collections();
-    }).then(collections => {
+  // Delete all data known to this adatper. Used for testing.
+  deleteAllSchemas() {
```
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: the naming is somewhat misleading, not deleting all schemas from the schemas collection but actually deleting all objects from the database.

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js [hunk 3]
```diff
@@ -81,15 +84,24 @@ export class MongoStorageAdapter {
...
+
+  _allCollections() {
+    return this.connect()
+    .then(() => this.database.collections())
+    .then(collections => {
       return collections.filter(collection => {
         if (collection.namespace.match(/\.system\./)) {
           return false;
         }
+        //TODO: If you have one app with a collection prefix that happens to be a prefix of another
```
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: in that case, that would mean we store objects in the same collections for both apps. A lot of things are gonna go bad before that :)

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js (hunk 2 above)
Comment (drew-gross) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: I think from the DB adapter point of view this is the correct name. SQL dbs, for example, won't have a schemas collection, and deleting the schema is actually the same thing as deleting all the data.

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js (hunk 3 above)
Comment (drew-gross) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: I don't think so. Imagine two app connecting to the DB, one with collectionPrefix = `abc` and the other with collectionPrefix = `abcdef` and you deleted all collections that have `abc` as the prefix, you will get all the collections from both apps.

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js (hunk 1 above)
Comment (drew-gross) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: Just so it's not exported? Sure.

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js (hunk 3 above)
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: hum right... we don't put a separator there right?

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js (hunk 2 above)
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: I would just have called it dropAll, but that's find

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js (hunk 1 above)
Comment (flovilmart) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: Thx!

File: src/Adapters/Storage/Mongo/MongoStorageAdapter.js (hunk 3 above)
Comment (drew-gross) [inline src/Adapters/Storage/Mongo/MongoStorageAdapter.js]: Nope =/ probably should have but that door closed years ago.

**Pull request merged**